include/gatekeeper_ggu.h

```diff
@@ -19,11 +19,57 @@
 #ifndef _GATEKEEPER_GGU_H_
 #define _GATEKEEPER_GGU_H_
 
+#include "gatekeeper_net.h"
+#include "gatekeeper_flow.h"
+
 /* Configuration for the GK-GT Unit functional block. */
 struct ggu_config {
-	unsigned int	lcore_id;
+	unsigned int      lcore_id;
+
+	/* The UDP source and destination port numbers for GGU. */
+	uint16_t          ggu_src_port;
+	uint16_t          ggu_dst_port;
+
+	/*
+	 * The fields below are for internal use.
+	 * Configuration files should not refer to them.
+	 */
+
+	/* RX queue on the front interface. */
+	uint16_t          rx_queue_front;
+	struct net_config *net;
+	struct gatekeeper_rss_config	rss_conf;
+};
+
+/*
+ * Since the length of IPv6 policy is much larger than IPv4 policy,
+ * to save network bandwidth, we choose to use different data structures
+ * to store the in-packet policies for IPv4 and IPv6, respectively.
+ *
+ * Packet format: Ethernet headers + IP header + UDP header + Data.
+ * In the IP payload, the format is: num_policies (8 bits) + proto1 + proto2
+ * + ... + policy1 + policy2 + ... .
+ */
+struct ggu_policy {
+	struct ip_flow flow;
+
+	struct {
+		/*
+		 * XXX Add state fields for the flow if necessary.
+		 * The policy decision sent to a GK block must have
+		 * enough information to fill out the fields of
+		 * struct flow_entry at the corresponding state.
+		 */
+		uint8_t  state;
+		uint32_t tx_rate_kb_cycle;
+		uint32_t grantor_id;
+		uint64_t cap_expire_at;
```

Comment:
What's the unit of this field (i.e. `cap_expire_at`)? It can't be the same unit of `cap_expire_at` in `struct flow_entry` because the later only makes sense at the machine where it was obtained.